app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorRequestView.java

```diff
@@ -186,82 +187,103 @@ private void requestFavor() {
   private void setFavorActivatedView(View v) {
     confirmFavorBtn.setVisibility(View.INVISIBLE);
     editFavorBtn.setVisibility(View.VISIBLE);
-    editFavorBtn.setText(getString(R.string.edit_favor));
     cancelFavorBtn.setVisibility(View.VISIBLE);
-    toggleTextViewsEditable(false);
-    updateViewFromStatus();
-    CommonTools.hideKeyboardFrom(Objects.requireNonNull(getContext()), v);
   }
 
   /** When edit button is clicked */
   private void startUpdatingFavor() {
-    cancelFavorBtn.setEnabled(true);
-    addPictureFromFilesBtn.setEnabled(true);
-    addPictureFromCameraBtn.setEnabled(true);
-    editFavorBtn.setText(R.string.confirm_favor_edit);
-    toggleTextViewsEditable(true);
+    currentFavor.setStatusId(Favor.Status.EDIT);
+    updateViewFromStatus(getView());
   }
 
   /** Gets called once favor has been updated on view. */
   private void confirmUpdatedFavor() {
-    getFavorFromView();
+    getFavorFromView(Favor.Status.REQUESTED);
     // update lists
-    ((MainActivity) Objects.requireNonNull(getActivity()))
-        .activeFavors.put(currentFavor.getId(), currentFavor);
-    ((MainActivity) Objects.requireNonNull(getActivity()))
-        .archivedFavors.remove(currentFavor.getId());
-    setFavorActivatedView(getView());
+    updateMainActivityLists(true);
+    updateViewFromStatus(getView());
+    // TODO: add db call
     showSnackbar(getString(R.string.favor_edit_success_msg));
   }
 
   /** Updates favor on DB. Updates maps on main activity hides keyboard shows snackbar */
   private void cancelFavor() {
-    currentFavor.updateStatus(Favor.Status.CANCELLED_REQUESTER);
-    MainActivity mainActivity = (MainActivity) getActivity();
-    assert mainActivity != null;
-    mainActivity.archivedFavors.put(currentFavor.getId(), currentFavor);
-    mainActivity.activeFavors.remove(currentFavor.getId());
-    CommonTools.hideKeyboardFrom(
-        Objects.requireNonNull(getContext()), Objects.requireNonNull(getView()));
-    updateViewFromStatus();
+    currentFavor.setStatusId(Favor.Status.CANCELLED_REQUESTER);
+    updateMainActivityLists(false);
+    updateViewFromStatus(getView());
     // Show confirmation and minimize keyboard
     showSnackbar(getString(R.string.favor_cancel_success_msg));
+    // TODO: insert db call
+  }
+
+  private void updateMainActivityLists(boolean favorIsActive) {
+    MainActivity mainActivity = (MainActivity) getActivity();
+    assert mainActivity != null;
```

Comment:
Try to avoid assert when possible, I would simply check for null in this case or use Objects.RequireNoNNull()

Reply:
updated